packages/big-design/src/components/Tooltip/Tooltip.tsx

```diff
@@ -13,90 +14,80 @@ export interface TooltipProps extends React.HTMLAttributes<HTMLDivElement> {
   inline?: boolean;
 }
 
-interface State {
-  visible: boolean;
-}
+export const Tooltip: React.FC<TooltipProps> = typedMemo(
+  ({ children, inline = true, modifiers, trigger, ...props }) => {
+    const [isVisible, setIsVisible] = useState(false);
+    const [tooltipContainer, setTooltipContainer] = useState<HTMLDivElement | null>(null);
+
+    useEffect(() => {
+      const container = document.createElement('div');
 
-export class Tooltip extends React.PureComponent<TooltipProps, State> {
-  static defaultProps: Partial<TooltipProps> = {
-    placement: 'top',
-    inline: true,
-  };
+      document.body.appendChild(container);
+      setTooltipContainer(container);
+    }, []);
 
-  state = {
-    visible: false,
-  };
+    useEffect(() => {
+      return () => {
+        if (tooltipContainer) {
+          document.body.removeChild(tooltipContainer);
+        }
+      };
+    }, [tooltipContainer]);
 
-  private tooltipContainer?: HTMLDivElement;
+    const renderContent = useMemo(() => {
+      return typeof children === 'string' ? <Small color="white">{children}</Small> : children;
+    }, [children]);
 
-  componentDidMount() {
-    this.tooltipContainer = document.createElement('div');
-    document.body.appendChild(this.tooltipContainer);
-  }
+    const hideTooltip = () => {
+      setIsVisible(false);
+    };
 
-  componentWillUnmount() {
-    if (this.tooltipContainer) {
-      document.body.removeChild(this.tooltipContainer);
-    }
-  }
+    const showTooltip = () => {
+      setIsVisible(true);
+    };
 
-  render() {
-    const { children, trigger, inline } = this.props;
+    const onKeyDown = (event: React.KeyboardEvent) => {
+      if (event.key === 'Escape') {
+        setIsVisible(false);
+      }
+    };
 
     return (
       <Manager>
         <Reference>
           {({ ref }) => (
             <StyledTooltipTrigger
               inline={inline}
-              onBlur={this.hideTooltip}
-              onFocus={this.showTooltip}
-              onKeyDown={this.onKeyDown}
-              onMouseEnter={this.showTooltip}
-              onMouseLeave={this.hideTooltip}
+              onBlur={hideTooltip}
+              onFocus={showTooltip}
+              onKeyDown={onKeyDown}
+              onMouseEnter={showTooltip}
+              onMouseLeave={hideTooltip}
               ref={ref}
             >
               {trigger}
             </StyledTooltipTrigger>
           )}
         </Reference>
-        {this.tooltipContainer
+        {tooltipContainer
           ? createPortal(
               <Popper
-                placement={this.props.placement}
-                modifiers={{ offset: { offset: '0, 8' }, ...this.props.modifiers }}
-                eventsEnabled={this.state.visible}
+                placement={props.placement || 'top'}
+                modifiers={{ offset: { offset: '0, 8' }, ...modifiers }}
```

Comment:
I had an issue with this `offset` object. React wanted me to memoize it. If you're having no issues with this I need to look at my code haha.

Reply:
This hasn't given me any issues *_so far_*.